roqoqo/src/noise_models/overrotation.rs

```diff
@@ -0,0 +1,442 @@
+type TwoQubitGateIndex = (String, (usize, usize));
+type TwoQubitOverrotation = Vec<(
+    TwoQubitGateIndex,
+    (
+        SingleQubitOverrotationDescription,
+        SingleQubitOverrotationDescription,
+    ),
+)>;
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(
+    feature = "json_schema",
+    derive(schemars::JsonSchema),
+    schemars(deny_unknown_fields)
+)]
+struct SingleQubitOverrotationOnGateSerialize {
+    /// Overrotation for single qubit gates.
+    single_qubit_overrotations: SingleQubitOverroation,
+    /// Overrotation for two qubit gates.
+    two_qubit_overrotations: TwoQubitOverrotation,
+}
+
+#[cfg(feature = "serialize")]
+impl From<SingleQubitOverrotationOnGate> for SingleQubitOverrotationOnGateSerialize {
+    fn from(value: SingleQubitOverrotationOnGate) -> Self {
+        let single_qubit_overrotations: SingleQubitOverroation =
+            value.single_qubit_overrotations.into_iter().collect();
+        let two_qubit_overrotations: TwoQubitOverrotation =
+            value.two_qubit_overrotations.into_iter().collect();
+        SingleQubitOverrotationOnGateSerialize {
+            single_qubit_overrotations,
+            two_qubit_overrotations,
+        }
+    }
+}
+
+#[cfg(feature = "serialize")]
+impl From<SingleQubitOverrotationOnGateSerialize> for SingleQubitOverrotationOnGate {
+    fn from(value: SingleQubitOverrotationOnGateSerialize) -> Self {
+        let single_qubit_overrotations: HashMap<
+            (String, usize),
+            SingleQubitOverrotationDescription,
+        > = value.single_qubit_overrotations.into_iter().collect();
+        let two_qubit_overrotations: HashMap<
+            (String, (usize, usize)),
+            (
+                SingleQubitOverrotationDescription,
+                SingleQubitOverrotationDescription,
+            ),
+        > = value.two_qubit_overrotations.into_iter().collect();
+
+        SingleQubitOverrotationOnGate {
+            single_qubit_overrotations,
+            two_qubit_overrotations,
+        }
+    }
+}
+
+impl SupportedVersion for SingleQubitOverrotationOnGate {
+    fn minimum_supported_roqoqo_version(&self) -> (u32, u32, u32) {
+        (1, 10, 0)
+    }
+}
+
+impl SingleQubitOverrotationOnGate {
+    /// Creates a new SingleQubitOverrotationOnGate with default values.
+    pub fn new() -> Self {
+        Self {
+            single_qubit_overrotations: HashMap::new(),
+            two_qubit_overrotations: HashMap::new(),
+        }
+    }
+
+    /// Sets overrotation for a single qubit gate.
+    ///
+    /// # Arguments
+    ///
+    /// * `gate` - The name of the gate.
+    /// * `qubit` - The qubit the gate acts on.
+    /// * `noise_description` - overrotation description for gate.
+    ///
+    /// # Returns
+    ///
+    /// `Self` - The overrotation model with the new overrotation on gate set.
+    pub fn set_single_qubit_overrotations(
```

Comment:
I've noticed that all the set/get functions are plural (`overrotations`). Is this correct?